pkg/experiment/ingester/client/client.go

```diff
@@ -127,16 +127,30 @@ func shouldBeHandledByCaller(err error) bool {
 	return true
 }
 
-// The default gRPC service config is explicitly set to
-// not retry and load balance between instances.
+// The default gRPC service config is explicitly set to balance between
+// instances.
 const grpcServiceConfig = `{
-    "methodConfig": [{
-        "name": [{"service": ""}],
-        "retryPolicy": {}
-    }],
-	"healthCheckConfig": {
-		"serviceName": "pyroscope.segment-writer"
-	}
+  "methodConfig": [
+    {
+      "name": [
+        {
+          "service": ""
+        }
+      ],
+      "retryPolicy": {
+        "maxAttempts": 4,
+        "initialBackoff": "0.1s",
+        "maxBackoff": "1s",
+        "backoffMultiplier": 2,
+        "retryableStatusCodes": [
+          "UNAVAILABLE"
+        ]
+      }
+    }
+  ],
+  "healthCheckConfig": {
+    "serviceName": "pyroscope.segment-writer"
+  }
 }`
```

Comment:
@kolesnikovae please take a look at this, I think after https://github.com/grpc/grpc-go/pull/7905 we cannot do this anymore using the service config.

Reply:
Thanks for pointing this out!

We do want to disable gRPC retries as we manage it on our own on the caller side.

I think we should just remove the `methodConfig`: 
```go
const grpcServiceConfig = `{
    "healthCheckConfig": {
         "serviceName": "pyroscope.segment-writer"
    }
}`
```

This was an extra measure to not enable retries via service config mistakenly as it may come, e.g. from DNS, xDS, etc. We actually disable retries explicitly when create the connection, so this is not needed (exp. if the client just ignores it)

Reply:
Yes that works!